src/lib/eventbridge.js

```diff
@@ -0,0 +1,183 @@
+const _ = require("lodash");
+const { ClearResult } = require("./utils");
+const retry = require("async-retry");
+const async = require("async");
+
+const regions = [
+	"us-east-1",
+	"us-east-2",
+	"us-west-1",
+	"us-west-2",
+	"ap-south-1",
+	"ap-northeast-1",
+	"ap-northeast-2",
+	"ap-southeast-1",
+	"ap-southeast-2",
+	"ca-central-1",
+	"eu-central-1",
+	"eu-west-1",
+	"eu-west-2",
+	"eu-west-3",
+	"eu-north-1",
+	"sa-east-1"
+];
+
+const deleteEventBridge = async (eventBridgeName, region, retryOpts, AWS) => {
+	const EventBridge = new AWS.EventBridge({ region });
+	await retry(async bail => {
+		try {
+			// List all rules
+			const rules = await getAllRulesBelongingToEventBridge(
+				eventBridgeName,
+				region,
+				AWS
+			);
+			for (const rule of rules) {
+				const targets = await getAllTargetsBelongingToRule(
+					eventBridgeName,
+					rule.name,
+					region,
+					AWS
+				);
+				const targetIDs = targets.map(target => target.targetId);
+				await EventBridge.removeTargets({
+					Ids: targetIDs,
+					Rule: rule.name,
+					EventBusName: eventBridgeName
+				}).promise();
+				await EventBridge.deleteRule({
+					Name: rule.name,
+					EventBusName: eventBridgeName
+				}).promise();
+			}
+			await EventBridge.deleteEventBus({
+				Name: eventBridgeName
+			}).promise();
+		} catch (e) {
+			if (e.code !== "Throttling") {
```

Comment:
have you verified this is the code? the other services usually have a longer code for throttling

Reply:
I'm not dealing here with throttling of each part of the eventbus (bus, rules & targets), but wrapping everything under the same umbrella.

This retry works quite nice for me in CF deletion